### ISSUE TITLE
Questions about /pause and /resume endpoints.

### ISSUE THREAD

**Issue by eacdy** [spring-cloud/spring-cloud-commons#275]
I'm using **Spring Cloud Edgware.RELEASE**, and I have some question about `/pause`  and `/resume`  endpoint.
When the Configuration is like below:

```yaml
endpoints:
  pause:
    enabled: true
```
**When I request these endpoints, I Can ONLY get a error code 404.**

BUT, When I configuration like this:
```yaml
endpoints:
  pause:
    enabled: true
  restart:
    enabled: true
```
Both endpoints Can work well!

When I dive into the code, I find code in `org.springframework.cloud.autoconfigure.RefreshEndpointAutoConfiguration.PauseResumeEndpoints`, like below:

```Java
@ConditionalOnEnabledEndpoint(value = "restart", enabledByDefault = false)
protected static class PauseResumeEndpoints {

  @Bean
  @ConditionalOnMissingBean
  @ConditionalOnEnabledEndpoint("pause")
  public RestartEndpoint.PauseEndpoint pauseEndpoint(
    RestartEndpoint restartEndpoint) {
    return restartEndpoint.getPauseEndpoint();
  }

  @Bean
  @ConditionalOnMissingBean
  @ConditionalOnEnabledEndpoint("resume")
  public RestartEndpoint.ResumeEndpoint resumeEndpoint(
    RestartEndpoint restartEndpoint) {
    return restartEndpoint.getResumeEndpoint();
  }

}
```

SO, I wonder why this class has annotation `@ConditionalOnEnabledEndpoint(value = "restart", enabledByDefault = false)` ？

Do we must set `endpoints.restart.enabled=true` ,then Can we use `/resume` and `/pause` ? If so, What's the meanning of configuration  `endpoints.pause.enabled=true`  and `endpoints.resume.enabled=true` ?

COULD YOU HELP ME? THANKS A LOT.

**Comment by ryanjbaxter** [comment]
This is the issue for the change https://github.com/spring-cloud/spring-cloud-commons/issues/117

@matosf could you potentially explain why the endpoints are conditional on `restart` being enabled?  I didnt understand the reasoning for that in the issue or the PR.

**Comment by matosf** [comment]
@ryanjbaxter was some time ago, but look at the the PR I believe the reasoning was since in the [docs](http://cloud.spring.io/spring-cloud-static/spring-cloud.html#_endpoints) it mentions restart is disabled by default and pause/resume endpoints give IMO similar functionality (i.e. pause+resume=restart), then it made sense to me that pause/resume endpoints enabling would be dependent of enabling restart.
 If not, they should be separated, and not inside a comon PauseResumeEndpoints class.

**Comment by brenuart** [comment]
A few comments about the doc indeed:
- http://cloud.spring.io/spring-cloud-static/Edgware.RELEASE/single/spring-cloud.html#_endpoints says the `Restart` endpoint is disabled by default (and doesn't say anything about whether resume/pause are enabled)
- the page listing the SpringBoot application properties (http://cloud.spring.io/spring-cloud-static/Edgware.RELEASE/single/spring-cloud.html#_appendix_compendium_of_configuration_properties) mentions `endpoints.restart.enabled` is `true` by default (ie *the opposite*).

I imagine this will be fixed once you address this issue... ;-)

**Comment by eacdy** [comment]
@brenuart I still think it is a bug.
I don't think `/pause` and `/restart` should have any quantum entanglement.

**Comment by ryanjbaxter** [comment]
I talked about this with the team today and we came to the conclusion that at the very least we should document the fact that in order to enable the `/pause` and `/restart` endpoints we should document that they are tied to the `/restart` endpoint and that it would need to be enabled first.

**Comment by ryanjbaxter** [comment]
Closed via cb93e9b

**Comment by sailor-Wang** [comment]
pause接口需要配置endpoints.restart.enabled=true确实让人费解。更直接的不应该是endpoints.pause.enabled=true 和 endpoints.resume.enabled=true吗
